diff -uNr /home/fiam/kdepim-3.3.0.orig/libkdepim/Makefile.am ./libkdepim/Makefile.am
--- /home/fiam/kdepim-3.3.0.orig/libkdepim/Makefile.am	2004-08-05 21:27:05.000000000 +0200
+++ ./libkdepim/Makefile.am	2004-09-11 14:28:18.001775560 +0200
@@ -24,7 +24,7 @@
   overlaywidget.cpp progressmanager.cpp progressdialog.cpp \
   statusbarprogresswidget.cpp ssllabel.cpp completionordereditor.cpp \
   resourceabc.cpp diffalgo.cpp addresseediffalgo.cpp calendardiffalgo.cpp \
-  htmldiffalgodisplay.cpp broadcaststatus.cpp
+  htmldiffalgodisplay.cpp broadcaststatus.cpp part.cpp
 
 noinst_HEADERS = categoryselectdialog.h categoryeditdialog.h ktimeedit.h 
 $(srcdir)/categoryselectdialog.h: categoryselectdialog_base.h
@@ -36,7 +36,7 @@
 AddressBookServiceIface_DIR = $(srcdir)/interfaces
 
 libkdepim_la_LDFLAGS = $(all_libraries) -no-undefined -version-info 1:0:0
-libkdepim_la_LIBADD  = $(LIB_KIO) $(LIB_KABC) \
+libkdepim_la_LIBADD  = $(LIB_KIO) $(LIB_KABC) $(LIB_KPARTS) \
   $(top_builddir)/libkcal/libkcal.la $(LIB_KDEUI) 
 #               $top_builddir)/libkdepim/resources/libkpimresources.la
 
diff -uNr /home/fiam/kdepim-3.3.0.orig/libkdepim/part.cpp ./libkdepim/part.cpp
--- /home/fiam/kdepim-3.3.0.orig/libkdepim/part.cpp	1970-01-01 01:00:00.000000000 +0100
+++ ./libkdepim/part.cpp	2004-09-11 14:27:58.791695936 +0200
@@ -0,0 +1,35 @@
+/*
+    This file is part of libkdepim.
+
+    Copyright (c) 2004 Tobias Koenig <tokoe@kde.org>
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Library General Public
+    License as published by the Free Software Foundation; either
+    version 2 of the License, or (at your option) any later version.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Library General Public License for more details.
+
+    You should have received a copy of the GNU Library General Public License
+    along with this library; see the file COPYING.LIB.  If not, write to
+    the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+    Boston, MA 02111-1307, USA.
+*/
+
+#include "part.h"
+
+using namespace KPIM;
+
+Part::Part( QObject *parent, const char *name )
+  : KParts::ReadOnlyPart( parent, name )
+{
+}
+
+Part::~Part()
+{
+}
+
+#include "part.moc"
diff -uNr /home/fiam/kdepim-3.3.0.orig/libkdepim/part.h ./libkdepim/part.h
--- /home/fiam/kdepim-3.3.0.orig/libkdepim/part.h	1970-01-01 01:00:00.000000000 +0100
+++ ./libkdepim/part.h	2004-09-11 14:27:58.798694872 +0200
@@ -0,0 +1,51 @@
+/*
+    This file is part of libkdepim.
+
+    Copyright (c) 2004 Tobias Koenig <tokoe@kde.org>
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Library General Public
+    License as published by the Free Software Foundation; either
+    version 2 of the License, or (at your option) any later version.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Library General Public License for more details.
+
+    You should have received a copy of the GNU Library General Public License
+    along with this library; see the file COPYING.LIB.  If not, write to
+    the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+    Boston, MA 02111-1307, USA.
+*/
+
+#ifndef KPIM_PART_H
+#define KPIM_PART_H
+
+#include <kparts/part.h>
+
+namespace KPIM {
+
+/**
+  This special part should be used as base class for all applications
+  which shall be integrated in Kontact.
+ */
+class Part : public KParts::ReadOnlyPart
+{
+  Q_OBJECT
+
+  public:
+    Part( QObject *parent, const char *name );
+    virtual ~Part();
+
+  signals:
+    /**
+      Emit this signal whenever the plugin with this part shall be
+      raised inside Kontact.
+     */
+    void raise();
+};
+
+}
+
+#endif
